src/js/components/TextInput/TextInput.js

```diff
@@ -308,12 +331,12 @@ const TextInput = forwardRef(
                             // we stole the focus, give it back
                             inputRef.current.focus();
                             closeDrop();
-                            if (onSelect) {
+                            if (handleSuggestionSelect.current) {
                               event.persist();
                               const adjustedEvent = event;
                               adjustedEvent.suggestion = suggestion;
-                              adjustedEvent.target = inputRef.current;
-                              onSelect(adjustedEvent);
+                              adjustedEvent.target = (ref || inputRef).current;
```

Comment:
Don't need `ref ||`, `useForwardedRef` already does the right thing

Reply:
changes made